### PR TITLE
CHE-1060: configure 'java.source.folder' attribute from the project wizard

### PR DESCRIPTION
**Pull request by svor**
@vparfonov take a look please. 
[Yo can see the result here! ](https://youtu.be/hHDK9usXoAM)

### CODE REVIEW

**Comment by TylerJewell** [comment]
1. Project type should be labled "Java" not "Plain Java Project" or "Plain Java".
2. We should not have any default package for projects created from this wizard.  Empty project, not "com.company".

**Comment by benoitf** [comment]
what if I have multiple source folders in my project that I import ?

like
api/src
client/src

**Comment by slemeur** [comment]
@TylerJewell : it has been handled into a separate PR: https://github.com/eclipse/che/pull/1313

File: plugins/plugin-java/che-plugin-java-plain/che-plugin-java-plain-ide/src/main/java/org/eclipse/che/plugin/java/plain/client/wizard/PlainJavaPageViewImpl.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    public void changeBrowseBtnVisibleState(boolean isVisible) {
+        browseButton.setVisible(isVisible);
+    }
+
+    @Override
+    public void changeSourceFolderFieldState(boolean isEnable) {
+        sourceFolderField.setEnabled(isEnable);
+    }
+
+    interface MavenPageViewImplUiBinder extends UiBinder<FlowPanel, PlainJavaPageViewImpl> {
```
Comment (azatsarynnyy) [inline plugins/plugin-java/che-plugin-java-plain/che-plugin-java-plain-ide/src/main/java/org/eclipse/che/plugin/java/plain/client/wizard/PlainJavaPageViewImpl.java]: PlainJavaPageViewImplUiBinder

**Comment by benoitf** [comment]
So can I work with
https://github.com/santanche/java2learn 
only one src folder but there is a lib folder to configure as well when we import project

these should work 
https://github.com/moleksyuk/Java-Design-Patterns
https://github.com/winterbe/java8-tutorial
(only one src folder)

here are two src folders
https://github.com/Nirman-Rathod/Java

**Comment by vparfonov** [comment]
@slemeur in case several source folder we need to your UI mock up

**Comment by benoitf** [comment]
I may have to specify as well all .jar files to get as libraries as well (or specify lib folder which will include all .jars from this directory) (multiple jars and/or multiple lib folder)

**Comment by gazarenkov** [comment]
As for the goal of the issue LGTM

**Comment by vparfonov** [comment]
ok

**Comment by benoitf** [comment]
it's pretty useless without being able to specify multi folder and without being able to pickup java libraries as well even.
While issue is only targeting one folder I think epic https://jira.codenvycorp.com/browse/CHE-915 could not be closed (add remaining part as new issue) until libraries and multiple source folders are handled with the wizard

**Comment by svor** [comment]
@benoitf I've created two issues for the configuring [library](https://jira.codenvycorp.com/browse/CHE-1207) and [source folders](https://jira.codenvycorp.com/browse/CHE-1207)

**Comment by benoitf** [comment]
@svor that's perfect

**Pull request merged**